src/models/converters/safes.rs

```diff
@@ -1,10 +1,24 @@
-use crate::models::service::safes::{AddressEx, SafeInfoEx};
+use crate::models::service::safes::{AddressEx, ImplementationVersionState, SafeInfoEx};
 use crate::providers::ext::InfoProviderExt;
 use crate::providers::info::{InfoProvider, SafeInfo};
+use semver::Version;
+use std::cmp::Ordering;
 
 // We need to add Sync as trait bound as info_provider moves across threads
 impl SafeInfo {
     pub async fn to_safe_info_ex(&self, info_provider: &(impl InfoProvider + Sync)) -> SafeInfoEx {
+        let min_chain_version = info_provider
+            .chain_info()
+            .await
+            .ok()
```

Comment:
Why do we not force unwrap this? The general assumption should be: if we cannot fetch the chain info we should not proceed, as we do not know if it is a valid chain.

Reply:
I will use `expect` in the following PR so we get some logged message in the unlikely case this would fail.